src/ts/components/CoinManageComponent/CoinInputComponent.ts

```diff
@@ -0,0 +1,42 @@
+import { generateRandomCoins, checkValidChargeMoney } from '../../utils/utils';
+import { emit, renderSnackBar, $, on } from '../../dom';
+
+export default class CoinInputComponent {
+  private $coinInput = $(
+    '.charge-form-section__coin-input'
+  ) as HTMLInputElement;
+  private $chargeButton = $(
+    '.charge-form-section__button'
+  ) as HTMLButtonElement;
+  private $totalCoin: HTMLElement = $('.charge-form-section__total-coin');
+  private $snackBarContainer: HTMLElement = $('.snack-bar-container');
+
+  constructor(private vendingMachineCoinManager) {
+    on(this.$chargeButton, 'click', this.onSubmitChargeButton);
+  }
+
+  private onSubmitChargeButton = (e: Event): void => {
+    e.preventDefault();
+
+    try {
+      checkValidChargeMoney(this.$coinInput.valueAsNumber);
+      this.vendingMachineCoinManager.addCoins(
+        generateRandomCoins(this.$coinInput.valueAsNumber)
+      );
+
+      this.$totalCoin.textContent =
+        this.vendingMachineCoinManager.getTotalAmount();
+
+      emit(this.$chargeButton, '@chargeInputSubmit', {
+        detail: {
+          coins: this.vendingMachineCoinManager.getCoins(),
+        },
+      });
+
+      this.$coinInput.value = '';
+      this.$coinInput.focus();
```

Comment:
저였다면 아래와 같이 했을 것 같습니다. 질문 주신 내용과도 겹치네요!
```suggestion
// CoinInputComponent.ts
onSubmitChargeButton = (e: Event): void => {
  e.preventDefault();

  try {
    const chargeMoney = this.$coinInput.valueAsNumber;
    // 1. 금액 자체의 유효성체크(10단위인지, 최대/최소값 범위에 있는지 등등)하고 현재 잔액에 더할 수 있는 값인지 체크
    checkValidChargeMoney(chargeMoney); 
    // 2. 유효한 값인 chargeMoney 를 더한다.
    this.vendingMachineCoinManager.addCoins(chargeMoney);

    // ui 변경
    this.$totalCoin.textContent = this.vendingMachineCoinManager.getTotalAmount();
    this.$coinInput.value = '';
    this.$coinInput.focus();
    
    // 이벤트 전파
    emit(this.$chargeButton, '@chargeInputSubmit', {
      detail: {
        coins: this.vendingMachineCoinManager.getCoins(),
      },
    });
  } catch(error) {
    ...
  }
}
      
// vendingMachineCoinManager.ts
addCoins(chargeMoney: number) {
  const coins = generateRandomCoins(chargeMoney);
        
  Object.entries(coins).forEach(([coin, count]: [string, number]) => {
    this.coins[coin] += count;
  });
}
```

Reply:
전체적으로 coin, chargeMoney, totalAmount 세 가지 의미가 혼용되어 사용되고 있는데요, 그래서 가독성도 좋지 않고 어떤 값을 의미하는지도 잘 모르겠습니다. 예를 들면 coinInput 이 아니라 chargeMoneyInput 이 아닐까요?
동전으로 바꾸기 전 잔돈 개념은 chargeMoney, 동전은 coin 으로 통일하면 좋겠네요!

그런 의미에서 CoinInputComponent 가 맞는지도 생각해보시면 좋겠네요!

Reply:
coinInput -> chargeMoneyInput 이 더 명확하고 직관적인 네이밍 같습니다. 이와 같은 부분들은 동일하게 통일해보도록 하겠습니다.

작성해 주신 예시 코드 감사합니다!
여기서 궁금한 점이 있어 추가로 질문드리도록 하겠습니다.
작성해주신 코드를 보면 `generateRandomCoins`는 vendingMachineCoinManager에서 해주는 것 같은데, 이 작업을 도메인에서 하게끔 하시는 이유가 있을까요..?! 😯

Reply:
코인을 생성해서 더한다 라는 문맥 상 addCoins() 가 하는 일이어야 하지 않을까! 싶어서 manager 에 두었습니다.
view 는 단순 input 의 value 에 대해 유효성 검사를 할 뿐 도메인이 해도 될 일들을 너무 떠안지는 않았으면 좋겠네요!

Reply:
답변 감사합니다. 너무 많은 일을 view에서 떠안지 않아도 된다...! 😊